pkg/pex/pex.go

```diff
@@ -35,22 +36,24 @@ import (
 // these have been carefully selected to work in a broad range of applications
 // and micro-optimizations are likely to be counterproductive.
 type PeerExchange struct {
-	ctx    context.Context
-	cancel context.CancelFunc
-	log    log.Logger
+	log log.Logger
 
 	h         host.Host
 	newParams func(ns string) GossipConfig
 
-	t     time.Time
-	as    map[string]advertiser
-	thunk chan<- func()
+	t            time.Time
+	as           map[string]advertiser
+	done         <-chan struct{}
+	thunk        chan<- func()
+	c            io.Closer
```

Comment:
PeX is a relatively big struct, I think it's important to have meaningful names. I would rename:
`c` --> `closer`
`as` --> `advertisers`